CoroutineTemplate/domain/build.gradle

```diff
@@ -1,41 +1,19 @@
-apply plugin: 'com.android.library'
-apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-parcelize'
-
-android {
-    compileSdkVersion android_compile_sdk_version
-    defaultConfig {
-        minSdkVersion android_min_sdk_version
-        targetSdkVersion android_target_sdk_version
-        versionCode android_version_code
-        versionName android_version_name
-
-        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
-        consumerProguardFiles "consumer-rules.pro"
-    }
+plugins {
+    id 'java-library'
+    id 'kotlin'
+}
 
-    buildTypes {
-        release {
-            minifyEnabled false
-            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
-        }
-    }
-    compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
-    }
-    kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
-    }
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
 }
 
 dependencies {
-    implementation(
-        project(':data'),
-
-        "androidx.core:core-ktx:$androidx_core_ktx_version",
-        "androidx.security:security-crypto:$android_crypto_version",
-        "com.google.dagger:hilt-android:$hilt_version",
-        "org.jetbrains.kotlinx:kotlinx-coroutines-android:$kotlinx_coroutines_version"
+    api(
+        "com.squareup.moshi:moshi-kotlin:$moshi_version",
+        "com.squareup.retrofit2:converter-moshi:$retrofit_version",
+        "com.squareup.moshi:moshi:$moshi_version",
```

Comment:
I have a feeling that we wouldn't want to expose `moshi` or `retrofit` to the upper level 🤔  I mean - domain will handle all the related jobs to the API and returning response to the `:data` to continue on processing right? So no module actually needs to know about these libs - hence `implementation` would work imo. Please correct me if I missed anything

Reply:
@sleepylee You're correct, each module has its responsibility, then we don't need to expose its adapter. 
Let me correct it.